fontforge/parsettf.c

```diff
@@ -6329,8 +6329,11 @@ SplineFont *SFReadTTF(char *filename, int flags, enum openflags openflags) {
         chosenname[strlen(chosenname)-1] = '\0';
     }
     ttf = fopen(strippedname,"rb");
-    if ( ttf==NULL )
-return( NULL );
+    if ( ttf==NULL ) {
+        if ( strippedname!=filename ) free(strippedname);
+        if ( chosenname!=NULL ) free(chosenname);
+        return( NULL );
+    }
```

Comment:
Seems preferable to set sf to NULL at the top and do:
```
    if ( ttf != NULL ) {
        sf = _SFReadTTF(ttf,flags,openflags,strippedname,chosenname,NULL);
        fclose(ttf);
    }
    if ( strippedname!=filename ) free(strippedname);
    if ( chosenname!=NULL ) free(chosenname);
    return( sf );
```
Rather than repeating the code.

Reply:
Thank you for your feedback! I've just submitted a new commit to take your remark into account.